python/snewpy/neutrino.py

```diff
@@ -37,50 +47,65 @@ def is_neutrino(self):
-        # The reported precision is not significant given current
-        # uncertainties, but is useful for comparing to the table of
-        # parameters presented on nu-fit.org.
-        if self.mass_order == MassHierarchy.NORMAL:
-            # Note: in NH, the mass splittings are: m1..m2..............m3.
-            self.theta12 = 33.44 * u.deg
-            self.theta13 =  8.57 * u.deg
-            self.theta23 = 49.20 * u.deg
-            self.deltaCP = 197 * u.deg
-            self.dm21_2  = 7.42e-5 * u.eV**2
-            self.dm32_2  = 2.517e-3 * u.eV**2
+    #mixing angles
+    theta12: u.Quantity[u.deg]
+    theta13: u.Quantity[u.deg]
+    theta23: u.Quantity[u.deg]
+    #CP violation phase         
+    deltaCP: u.Quantity[u.deg]
+    #square mass difference
+    dm21_2: u.Quantity[u.eV**2]
+    dm32_2: Optional[u.Quantity] = None
+    dm31_2: Optional[u.Quantity] = None
+    #mass ordering
+    mass_order: Optional[MassHierarchy] = None
+    # Note: in IH, the mass splittings are: m3..............m1..m2.
+    
+    def _repr_markdown_(self):
+        s = [f'**{self.__class__.__name__}**']
+        s+=  ['|Parameter|Value|',
+              '|:--------|:----:|']
+        for name, v in self.__dict__.items():
+            try: 
+                s += [f"|{name}|{v._repr_latex_()}"]
+            except:
+                try: 
+                    s += [f"|{name}|{v.name}"]
+                except:
+                    s += [f"|{name}|{v}|"]
+        return '\n'.join(s)
+        
+    def __post_init__(self):
+        #calculate the missing dm2
+        if self.dm31_2 is None: 
+            self.dm31_2 = self.dm32_2+self.dm21_2
+        if self.dm32_2 is None: 
+            self.dm32_2 = self.dm31_2-self.dm21_2  
+        #evaluate mass ordering
+        if self.mass_order is None:
+            self.mass_order = MassHierarchy.derive_from_dm2(*self.get_mass_square_differences())
+        #validate angles
+        #angles_sum = sum(self.get_mixing_angles())
+        #assert angles_sum==90<<u.deg, f'Mixing angles sum is {angles_sum}!=90 degrees'
+        #check hierarchy
+        if self.mass_order==MassHierarchy.NORMAL:
+            assert self.dm32_2>0, 'dm32_2 should be positive for NH'
+            assert self.dm31_2>0, 'dm31_2 should be positive for NH'
         else:
-            # Note: in IH, the mass splittings are: m3..............m1..m2.
-            self.theta12 = 33.45 * u.deg
-            self.theta13 =  8.60 * u.deg
-            self.theta23 = 49.30 * u.deg
-            self.deltaCP = 282 * u.deg
-            self.dm21_2  = 7.42e-5 * u.eV**2
-            self.dm31_2  = -2.498e-3 * u.eV**2
+            assert self.dm32_2<0, 'dm32_2 should be negative for IH'
+            assert self.dm31_2<0, 'dm31_2 should be negative for IH'
+        #validate dm2
+        dm2_sum = self.dm32_2+self.dm21_2-self.dm31_2
+        assert dm2_sum==0, f'dm32_2+dm31_2-dm31_2 = {dm2_sum} !=0'
```

Comment:
Is there a risk of running into numerical issues here? (E.g. note that `assert 0.1 + 0.1 + 0.1 == 0.3` raises an `AssertionError`.)

Reply:
You're right, thanks. I'll fix it with `np.isclose`